pkg/services/gitrepo/service.go

```diff
@@ -6,83 +6,11 @@ import (
 	"io/ioutil"
 	"os"
 
-	"github.com/weaveworks/weave-gitops/pkg/models"
-
 	"github.com/weaveworks/weave-gitops/pkg/git"
 	"github.com/weaveworks/weave-gitops/pkg/gitproviders"
 	"github.com/weaveworks/weave-gitops/pkg/logger"
 )
 
-type RepoWriter interface {
-	CreatePullRequest(ctx context.Context, info gitproviders.PullRequestInfo) error
-	WriteAndMerge(ctx context.Context, repoDir, commitMsg string, manifests []models.Manifest) error
-	CloneRepo(ctx context.Context, branch string) (func(), string, error)
-	GetDefaultBranch(ctx context.Context) (string, error)
-	CommitAndPush(ctx context.Context, commitMsg string, filters ...func(string) bool) error
-	CheckoutBranch(newBranch string) error
-	Write(ctx context.Context, path string, content []byte) error
-	Remove(ctx context.Context, path string) error
-}
-
-type RepoWriterSvc struct {
-	URL         gitproviders.RepoURL
-	GitProvider gitproviders.GitProvider
-	GitClient   git.Git
-	Logger      logger.Logger
-}
-
-var _ RepoWriter = &RepoWriterSvc{}
-
-func NewRepoWriter(url gitproviders.RepoURL, gitProvider gitproviders.GitProvider, gitClient git.Git, logger logger.Logger) RepoWriter {
-	return &RepoWriterSvc{URL: url, GitProvider: gitProvider, GitClient: gitClient, Logger: logger}
-}
-
-func (rw *RepoWriterSvc) CreatePullRequest(ctx context.Context, info gitproviders.PullRequestInfo) error {
-	pr, err := rw.GitProvider.CreatePullRequest(ctx, rw.URL, info)
-	if err != nil {
-		return fmt.Errorf("unable to create pull request: %w", err)
-	}
-
-	rw.Logger.Println("Pull Request created: %s\n", pr.Get().WebURL)
-
-	return nil
-}
-
-func (rw *RepoWriterSvc) WriteAndMerge(ctx context.Context, repoDir, commitMsg string, manifests []models.Manifest) error {
-	for _, m := range manifests {
-		if err := rw.GitClient.Write(m.Path, m.Content); err != nil {
-			return fmt.Errorf("Failed to write manifest: %w", err)
-		}
-	}
-
-	return rw.CommitAndPush(ctx, commitMsg, func(fname string) bool {
-		for _, m := range manifests {
-			if fname == m.Path {
-				return true
-			}
-		}
-
-		return false
-	})
-}
-
-func (rw *RepoWriterSvc) CloneRepo(ctx context.Context, branch string) (func(), string, error) {
-	return CloneRepo(ctx, rw.GitClient, rw.URL, branch)
-}
-
-func (rw *RepoWriterSvc) CommitAndPush(ctx context.Context, commitMsg string, filters ...func(string) bool) error {
-	return CommitAndPush(ctx, rw.GitClient, commitMsg, rw.Logger, filters...)
-}
-
-func (rw *RepoWriterSvc) CheckoutBranch(newBranch string) error {
-	err := rw.GitClient.Checkout(newBranch)
-	if err != nil {
-		return fmt.Errorf("error checking out branch %s, %w", newBranch, err)
-	}
-
-	return nil
-}
-
 func CommitAndPush(ctx context.Context, client git.Git, commitMsg string, logger logger.Logger, filters ...func(string) bool) error {
```

Comment:
since you removed the add cmd does anything still use this? I know we will still use the commit part of it but not the push part.

Reply:
Okay I did a quick check and upgrade is the only spot still using this. Upgrade will be getting updated soon anyway so its up to you if we treat it like install with a message or just leave it as is for now.